mlflow/store/artifact/artifact_repo.py

```diff
@@ -384,14 +384,14 @@ def upload_trace_data(self, trace_data: str) -> None:
 def write_local_temp_trace_data_file(trace_data: str):
     with tempfile.TemporaryDirectory() as temp_dir:
         temp_file = Path(temp_dir, TRACE_DATA_FILE_NAME)
-        temp_file.write_text(trace_data)
+        temp_file.write_text(trace_data, encoding="utf-8")
         yield temp_file
 
 
 def try_read_trace_data(trace_data_path):
     if not os.path.exists(trace_data_path):
         raise MlflowTraceDataNotFound(artifact_path=trace_data_path)
-    with open(trace_data_path) as f:
+    with open(trace_data_path, encoding="utf-8") as f:
```

Comment:
q: Is it backwards compatible? What would happen if a file with a different encoding is read as "utf8" here?